packages/itwin/tree-widget/src/test/trees/categories-tree/internal/CategoriesVisibilityHandler.test.ts

```diff
@@ -76,24 +88,40 @@ describe("CategoriesVisibilityHandler", () => {
     });
   }
 
-  async function createVisibilityTestData({ imodel, isVisibleOnInitialize }: { imodel: IModelConnection; isVisibleOnInitialize: boolean }) {
-    const commonProps = await createCommonProps(imodel, isVisibleOnInitialize);
-    const handler = new CategoriesVisibilityHandler(commonProps);
+  async function createVisibilityTestData({
+    imodel,
+    categoryIds,
+    testDataVisibilityInitializer,
+  }: {
+    imodel: IModelConnection;
+    testDataVisibilityInitializer?: TestDataVisibilityInitializer;
+    categoryIds: Id64Array;
+  }) {
+    const commonProps = await createCommonProps({ imodel, categoryIds });
+    const handler = new CategoriesVisibilityHandler({ idsCache: commonProps.idsCache, viewport: commonProps.viewport });
+    const modelsTreeHandler = createModelsTreeVisibilityHandler({
+      idsCache: commonProps.modelsTreeIdsCache,
+      viewport: commonProps.viewport,
+      imodelAccess: commonProps.imodelAccess,
+    });
```

Comment:
I don't think category tests should use models tree ids cache or visibility handler. All visibility changes should be made directly on the display system, to make sure our tests are targeted towards what we want to test and nothing more.